### DISCUSSION TITLE
Having trouble defining custom id3 tags instead of flexible attributes.

### DISCUSSION THREAD

**Discussion by bbaserdem** [beetbox/beets#4569] [Q&A]
I recently started using beets to organize my library. I am used to using picard; but the way I organize my library is that I overwrite the albumartist field of certain 'various artists' releases so that music players will group these albums separately. (For example, the record label [Chillhop Music](https://chillhop.bandcamp.com/) release a lot of compilations; but I make the album artist 'Chillhop Music' when I use picard so that music players, including mpd, will group these separately from my other various artist albums; which  would otherwise make compilation albums very difficult to find in list view.) I also use a custom id3 tag; called `series` which roughly correspond to [Series, specifically release group series,](https://musicbrainz.org/doc/Series) relationship on musicbrainz.

I want this to be a set tag with the information written as id3 tags to the individual tracks. I wrote my own plugin to do this, and currently the relevant lines are (there is [more here](https://github.com/bbaserdem/Homefiles/blob/main/.local/beets/beetsplug/sbp.py) but it's just some custom path format functions and fields.)

``` .local/beets/beetsplug/sbp.py
from beets.plugins import BeetsPlugin
import mediafile

class MyPlugin(BeetsPlugin):
    """Plugin object."""
    def __init__(self):
        """Self definition."""
        super().__init__()
        _series = mediafile.MediaField(
            mediafile.MP3DescStorageStyle('Series'),
            mediafile.StorageStyle('Series'),
            out_type=str
        )
        self.add_media_field('series', _series)

```

I know that this file is loaded well, because i include it in `pluginpaths`; and can load my `sbp` plugin just fine. (Other stuff I defined here works.) However; when I do `beet fields`; series shows as a flexible attribute both for albums and for items; and when I set the series for an album or for tracks, it's not set in the id3 tags in the files. (I have other workflow besides beets that make use of tags from music tracks, so I want to set them not only in the beets database but on the tracks as well.)

Currently, I'm stuck on writing the aforementioned data to the proper tracks; and I followed [the documentation](https://beets.readthedocs.io/en/stable/dev/plugins.html#extend-mediafile) for this. But I think I'm failing at something. I would appreciate because I'm not very well versed with python, or with beets.

In the future; I want to expand this plugin to; (not necessarily my focus; but if you have ideas on how to tackle this because I don't know where to start with these.)

* Pull the release group series relationship from musicbrainz database
* Prevent mbsync plugin from overwriting the albumartist if the musicbrainz database version is 'Various Artists'
* Stop the removal of the compilation flag on `beet write` if the albumartist is not 'Various Artists`.

**Comment by sampsyo** [comment]
Interesting! Just to check, are you doing `beet write` following a `beet modify series=...` or similar? Or does your configuration file have automatic writing enabled? Have you looked at the verbose logs for `beet write` applied to one of these files?

What tool are you using to look at the ID3 tags to confirm that the data is not there?

**Comment by bbaserdem** [comment]
I'm checking the id3 tags on picard; unrecognized tags should show up but they don't.

I have automatic writing enabled; issued write and it seems to be working? I don't know why it was not working before. I'm guessing because I set the tag on the album; and not on the items.

Issuing beet write now garbles the albumtype and albumtypes field; makes them from `album` to `['a', 'l', 'b', 'u', 'm']` so not willing to investigate much further.

**Reply by sampsyo** [reply]
For the later issue, please see #4528.
